### ISSUE TITLE
Freq Linear Regression - Allows categorical variables while it should not

### ISSUE THREAD

**Issue by JohnnyDoorn** [jasp-stats/jasp-issues#1029]
* JASP version: 0.14
* OS name and version: Ubuntu 20.04
* Analysis: Frequentist linear regression
* Bug description: categorical variables are able to be selected as predictor variables, and are treated as continuous.
* Expected behaviour: should not happen (in concordance with the icons shown in the boxes, see screenshot)

#### Steps to reproduce:
load a data set with continuous and categorical variables (e.g., debug), specify linear regression and select categorical variable as predictor, this should not be possible.

![Screenshot from 2020-10-29 17-29-02](https://user-images.githubusercontent.com/15704203/97603033-4edd7c80-1a0c-11eb-978d-a2c1402380e7.png)

**Comment by Kucharssim** [comment]
One could argue that this is fine, we should “only” implement handling categorical predictors 😅 https://github.com/jasp-stats/jasp-issues/issues/1022

**Comment by JohnnyDoorn** [comment]
Yes, but for now this is just misleading, I mean look at the example in the screenshot, I have a nominal variable with 4 levels a,b,d,d, and JASP just handles them as 1,2,3,4. As the quickest fix this should not be possible, but of course we should work on making it handle categorical variables well (as in, as an actual factor...)

**Comment by vandenman** [comment]
I think it's most important to be consistent across the Bayesian and frequentist analyses. Usually, linear regression with categorical predictors and continuous predictors is called ANCOVA. That's also what we recommend at the moment when people want to add categorical predictors to Bayesian linear regression, do a Bayesian ANCOVA instead.

@JohnnyDoorn Shall we fix this for the bugfix release in a couple of weeks by disallowing discrete variables? It's probably a lot more work to get the correct output for discrete input.

**Comment by JohnnyDoorn** [comment]
That sounds good. 
I do think that we should add categorical predictors to linear regression, because it has some different output compared to ANCOVA (like [this user](https://forum.cogsci.nl/discussion/6576/linear-regression-with-a-categorical-covariate#latest) was expecting from JASP). But yes, the model fit and everything is the same - see below for an illustration of the three different approaches:

```
y <- rnorm(20)
f <- factor(rep(letters[1:4], 5))
cont <- rnorm(20, y, 3)

summary(lm(y ~ as.numeric(f) + cont)) # This is what JASP currently does
summary(lm(y ~ f + cont))             # This is what you would expect 
summary(aov(y ~ f + cont))            # Same inference for the covariate, no separate coefficients for the factor levels
```

**Comment by Kucharssim** [comment]
> Yes, but for now this is just misleading,...

Sure, I agree that fixing this in the short term and thinking about how to implement the categorical predictors properly in the long term is what we should do. I just linked the other issue here just so that we don't forget that they are somewhat related 🙂 

> I think it's most important to be consistent across the Bayesian and frequentist analyses. Usually, linear regression with categorical predictors and continuous predictors is called ANCOVA.

I agree with Johnny there. Regardless of what we wish from consistency between Bayesian and classical analyses, people expect (classical) linear regression handling categorical predictors (which in my mind is perfectly reasonable), and some of them would resort to creating dummy variables to use in the regression rather than running ANCOVA (for the reasons Johnny listed above). I don't see a reason why we should not make it easier for our users.

**Comment by vandenman** [comment]
> people expect (classical) linear regression handling categorical predictors

I agree.

> I don't see a reason why we should not make it easier for our users.

For the frequentist analyses, we should definitely do this, ideally for 0.15 already.


The problem is the Bayesian analyses. `Bayesfactor` and `BAS` give different results if you include categorical predictors (unfortunately this is not a numerical issue). The problem is then that a Bayesian ANCOVA and a Bayesian linear regression yield different Bayes factors and they shouldn't. That's why categorical predictors are not allowed in Bayesian linear regression. At the moment, I don't know how to fix this (`BAS` is generally also very inefficient for categorical predictors).
